File: Programmers/124 나라의 숫자.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
	string answer = "";
	int res = 0;
	while (n > 0)
	{
        res = n;
		n /= 3;
		if (res % 3 == 2)
			answer = '2' + answer;
		else if (res % 3 == 1)
			answer = '1' + answer;
		else
		{
			answer = '4' + answer;
			n--;
		}

	}
	return answer;
}

File: Programmers/124 나라의 숫자.java
class Solution {
    public String solution(int n) {
        String answer = "";
        
        while(n > 0)
        {
            int res = n;
            n /= 3;
            if(res % 3 == 1)
                answer = "1" + answer;
            else if (res % 3 == 2)
                answer = "2" + answer;
            else
            {
                answer = "4" + answer;
                n--;
            }
        }
        
        
        return answer;
    }
}